Bottom: 108ab6afd619538c4707d9fcc73acb14eb29fbcf
Top:    3f518c50d8354809224e4ab4b34e2a670885a71a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 17:48:06 +0000

hid/common: Don't leak the dir resource in hid_load_dir()

Add missing closedir() call.

(Caught with cppcheck)


---

diff --git a/src/hid/common/hidinit.c b/src/hid/common/hidinit.c
index 0bdd395..3695f7d 100644
--- a/src/hid/common/hidinit.c
+++ b/src/hid/common/hidinit.c
@@ -117,6 +117,7 @@ hid_load_dir (char *dirname)
       free (path);
     }
   free (dirname);
+  closedir (dir);
 }
 
 void
